vllm/entrypoints/openai/serving_completion.py

```diff
@@ -150,15 +151,20 @@ async def create_completion(
                     log_tracing_disabled_warning()
 
                 if isinstance(sampling_params, BeamSearchParams):
-                    if not isinstance(self.engine_client, AsyncLLMEngine):
+                    if isinstance(self.engine_client, AsyncLLMEngine):
+                        generator = self.engine_client.beam_search(
+                            prompt_inputs["prompt_token_ids"], request_id_item,
+                            sampling_params)
+                    elif isinstance(self.engine_client, MQLLMEngineClient):
+                        generator = self.engine_client.beam_search(
+                            prompt_inputs["prompt_token_ids"], request_id_item,
+                            sampling_params, lora_request)
```

Comment:
Is beam search supported with LoRA?

It seems like it does not work through the `AsyncLLMEngine` since we aren't passing the `lora_request` parameter there. So I think you could collapse these two cases since `lora_request` will always be none when passed to `MQLLMEngine`